src/full_client.toit

```diff
@@ -295,30 +310,39 @@ abstract class ReconnectionStrategyBase implements ReconnectionStrategy:
       [--disconnect-transport]
       [--send-connect]
       [--receive-connect-ack]:
-    attempt-counter := -1
+    current-attempt-time := Time.monotonic-us
+    // If we try to do a connection rapidly after we managed to connect, then we
+    // got probably disconnected. In that case, treat this as if we didn't manage to
+    // connect and keep the last attempt counter to have some back-off.
+    if current-attempt-time - last-attempt-time_ > reset-duration_.in_us:
+      // Consider this a completely fresh attempt to connect.
+      attempt-counter_ = -1
```

Comment:
It feels a bit weird that this is not really the counter, but the count of the current attempt.

Reply:
Renamed to 'attempt'.
done.